src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -479,17 +465,17 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
                 (((long)binaryForm[offset + 6]) << 8) +
                 (((long)binaryForm[offset + 7])));
 
-            SubAuthorities = new int[binaryForm[offset + 1]];
+            int subAuthoritiesLength = binaryForm[offset + 1];
 
             //
             // Subauthorities are represented in big-endian format
             //
 
-            for (byte i = 0; i < binaryForm[offset + 1]; i++)
+            for (int i = 0; i < binaryForm[offset + 1]; i++)
             {
                 unchecked
```

Comment:
The `unchecked` can be removed.  Nothing in dotnet/runtime has checked enabled.

Reply:
Done.